src/queryCache.js

```diff
@@ -313,10 +313,11 @@ export function makeQueryCache() {
 
         // Do we need to retry the request?
         if (
-          // Only retry if the document is visible
-          query.config.retry === true ||
-          query.state.failureCount <= query.config.retry
+          query.config.retryChecker(error, query.state.failureCount) &&
+          (query.config.retry === true ||
+            query.state.failureCount <= query.config.retry)
```

Comment:
Reverse this order so that the checker isn't even called if it's not true or less than the retry count?